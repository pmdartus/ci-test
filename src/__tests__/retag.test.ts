import t from '../retag';

test('render simple element', () => {
    const root = t.h1();
    expect(root).toMatchSnapshot();
});

test('render simple element', () => {
    const root = t.h1(['Hello world'])
    expect(root).toMatchSnapshot();
});

test('render nested children', () => {
    const items = ['foo', 'bar', 'baz'];
    const root = t.ul(
        items.map(content => t.li([ content ]))
    );

    expect(root).toMatchSnapshot();
});

test('render attributes', () => {
    const root = t.h1({
        className: 'title',
        style: {
            color: 'red',
        }
    }, [
        'Hello world',
    ]);

    expect(root).toMatchSnapshot();
});