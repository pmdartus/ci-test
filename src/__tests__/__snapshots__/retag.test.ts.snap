// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render attributes 1`] = `
<h1
  class="title"
  style="color: red;"
>
  Hello world
</h1>
`;

exports[`render nested children 1`] = `
<ul>
  <li>
    foo
  </li>
  <li>
    bar
  </li>
  <li>
    baz
  </li>
</ul>
`;

exports[`render simple element 1`] = `<h1 />`;

exports[`render simple element 2`] = `
<h1>
  Hello world
</h1>
`;
